### PR TITLE
feat(migration): Add answer actions to the UI

### LINKED ISSUE #38: Migrate Question Detail Page - Answer
Reimplementation of answering modal with Remix hooks and the new controller implemented. #12

### PR DESCRIPTION
**Pull request by AldoDominguezWizeline**
#### What does this PR do?
It adds the following actions  in the question detail page:

- Add an answer to a question
- Edit your previously created answer to a question
- Delete your previously created answer to a question

#### Where should the reviewer start?
- app/components/Modals/AnswerModal/AnswerModal.jsx
- app/components/Modals/DeleteAnswerModal/DeleteAnswerModal.Styled.jsx
- app/routes/questions/$questionId.jsx
- app/components/QuestionDetail/QuestionDetail.jsx

#### How should this be manually tested?
(List of steps to reproduce, corroborate or tests to run. Write this section clear enough so that external users can also follow it and test the fix.)
1. As an admin, go to the question detail page of any question that hasn't been answered yet.
2. Click the 'ANSWER' option, in the displayed pop-up, enter your answer's text, then submit it.
3. The answer should be created and now displayed under the question in the question detail page.
4. Verify you can edit that same answer, your answer should be updated with the new text when you input a new value and submit it after clicking the pencil icon next to your previous answer in the question detail page.
5. Verify you can delete your previously created answer with the trash bin icon next to your answer in the question detail page.

#### Any background context you want to provide?
Some missing React components had to be added from the old front-end project.

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #38

#### Screenshots

- **Add an answer**

https://user-images.githubusercontent.com/104774638/198692658-8be6a769-d43a-4eb7-94dc-eca4a2c74aed.mov

- **Update an answer**

https://user-images.githubusercontent.com/104774638/198694415-04f55b19-18c0-48de-bd0b-46036cf96f96.mov

- **Delete an answer**

https://user-images.githubusercontent.com/104774638/198694450-3f2fc928-8489-4426-a9d2-a5441fc6e785.mov

### CODE REVIEW

File: app/components/QuestionDetail/QuestionDetail.jsx [hunk 1]
```diff
@@ -35,57 +34,67 @@ function QuestionDetails(props) {
...
+    showAnswerModal: false,
+    showAssignAnswerModal: false,
+  };
+
   const [state, setState] = useState(initialState);
   const [writingCommentOnMobile, setWritingCommentOnMobile] = useState(false);
 
   const { questionId } = useParams();
 
+  const setUrl = () => {
```
Comment (albertcastaned) [inline app/components/QuestionDetail/QuestionDetail.jsx:47]: @AldoDominguezWizeline  What is setUrl used for?

File: app/components/Atoms/Input/Input.Styled.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import styled from 'styled-components';
+//import '../../../styles/MediaQueries.scss';
```
Comment (albertcastaned) [inline app/components/Atoms/Input/Input.Styled.jsx:2]: Remove comments

**Review by albertcastaned — APPROVED** [review]
Nice job!!

File: app/components/Atoms/Input/Input.Styled.jsx (hunk 2 above)
Comment (AldoDominguezWizeline) [reply app/components/Atoms/Input/Input.Styled.jsx]: Nice catch, comment removed, thank you very much!

File: app/components/QuestionDetail/QuestionDetail.jsx (hunk 1 above)
Comment (AldoDominguezWizeline) [reply app/components/QuestionDetail/QuestionDetail.jsx]: I preserved that function from the previous react project, it seems it just replaces the navigation history in the browser from what I could gather, but I tested again and the function is not necessary, so I removed it, nice catch and thank you!

**Pull request merged**